### PR TITLE
Restructuring developer documentation for easier reach

### PR DESCRIPTION
**Pull request by subhramit**
`contributing.md` used to be an independent topic in the devdocs in itself. 
Apart from having its own original content as index, we propose to make it a parent that has two children: 
1. The "Getting into the code" section. This contains:
    1a."JabRef's development strategy", 
    1b."Setting up a local workspace" and 
    1c."High-level documentation"
  This used to be an independent section in itself, which will be shifted here.
2. Developer FAQs (Refer https://github.com/JabRef/jabref/pull/11456).

This makes important documentation more prominent to reach for newcomers.

This PR also fixes some grammatical errors in FAQs, and adds some pointers about reaching IntlliJ Settings in M2 Macs while setting up the development environment.

Update: Also adds the link to the developer FAQs in the github actions bot newcomer greeting message.
### Mandatory checks

- [ ] ~Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)~
- [ ] ~Tests created for changes (if applicable)~
- [ ] ~Manually tested changed features in running JabRef (always required)~
- [ ] ~Screenshots added in PR description (for UI changes)~
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: .github/workflows/add-greeting-to-issue.yml [hunk 1]
```diff
@@ -24,6 +24,8 @@ jobs:
 
           Newcomers, we're excited to have you on board. Start by exploring our [Contributing](https://github.com/JabRef/jabref/blob/main/CONTRIBUTING.md) guidelines, and don't forget to check out our [workspace setup guidelines](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace) to get started smoothly.
 
+          In case you encounter failing tests during development, don't forget to check our [developer FAQs](https://devdocs.jabref.org/contributing/faq.html)!
```
Comment (koppor) [inline .github/workflows/add-greeting-to-issue.yml:27]: Try to avoid negations: `please check our` instead of `don't forget to...`

**Comment by subhramit** [comment]
Moving to WIP till:
1. Links pointing to the moved parent topics are fixed
2. Screenshots are added in PR Description

**Comment by subhramit** [comment]
We will probably have to close this because: https://just-the-docs.com/docs/navigation-structure/#example-children-within-children
>Currently, the navigation structure is limited to 3 levels: grandchild pages cannot themselves have child pages.

**Comment by koppor** [comment]
Will be huge effort to finalize. Thus, better improve the Markdown files (as you did). We will do a larger restructuring at another time.

**Pull request closed without merging**